### PR TITLE
Implement IWAE estimator for EIG

### PR DESCRIPTION
**Pull request by ae-foster**
This introduces the IWAE which handily requires no new guides -- we just re-use old posterior guides with a new loss function. We can interpret this as an improvement on NMC.

### CODE REVIEW

File: pyro/contrib/glmm/glmm.py [hunk 1]
```diff
@@ -317,9 +316,9 @@ def bayesian_linear_model(design, w_means={}, w_sqrtlambdas={}, re_group_sizes={
             base_dist = dist.Normal(prediction_mean, obs_sd)
             # You can add loc via the linear model itself
             k = k.expand(prediction_mean.shape)
-            transforms = [AffineTransform(loc=torch.tensor(0.), scale=k), SigmoidTransform()]
-            tr_dist = dist.TransformedDistribution(base_dist, transforms)
-            response_dist = dist.CensoredDistribution(tr_dist, upper_lim=1.-epsilon, lower_lim=epsilon).independent(1)
+            response_dist = dist.CensoredSigmoidNormal(
+                loc=k * prediction_mean, scale=k * obs_sd, upper_lim=1. - epsilon, lower_lim=epsilon
+            ).independent(1)
```
Comment (ae-foster) [inline pyro/contrib/glmm/glmm.py:321]: avoid `nan` loss for sigmoid model
